src/TolerantDefinitionResolver.php

```diff
@@ -0,0 +1,880 @@
+<?php
+declare(strict_types = 1);
+
+namespace LanguageServer;
+
+use PhpParser\Node;
+use PhpParser\PrettyPrinter\Standard as PrettyPrinter;
+use phpDocumentor\Reflection\{Types, Type, Fqsen, TypeResolver};
+use LanguageServer\Protocol\SymbolInformation;
+use LanguageServer\Index\ReadableIndex;
+
+class TolerantDefinitionResolver implements DefinitionResolverInterface
+{
+    /**
+     * @var \LanguageServer\Index\ReadableIndex
+     */
+    private $index;
+
+    /**
+     * @var \phpDocumentor\Reflection\TypeResolver
+     */
+    private $typeResolver;
+
+    /**
+     * @var \PhpParser\PrettyPrinterAbstract
+     */
+    private $prettyPrinter;
+
+    /**
+     * @param ReadableIndex $index
+     */
+    public function __construct(ReadableIndex $index)
+    {
+        $this->index = $index;
+        $this->typeResolver = new TypeResolver;
+        $this->prettyPrinter = new PrettyPrinter;
+    }
+
+    /**
+     * Builds the declaration line for a given node
+     *
+     * @param Node $node
+     * @return string
+     */
+    public function getDeclarationLineFromNode(Node $node): string
+    {
+        if ($node instanceof Node\Stmt\PropertyProperty || $node instanceof Node\Const_) {
+            // Properties and constants can have multiple declarations
+            // Use the parent node (that includes the modifiers), but only render the requested declaration
+            $child = $node;
+            /** @var Node */
+            $node = $node->getAttribute('parentNode');
+            $defLine = clone $node;
+            $defLine->props = [$child];
+        } else {
+            $defLine = clone $node;
+        }
+        // Don't include the docblock in the declaration string
+        $defLine->setAttribute('comments', []);
+        if (isset($defLine->stmts)) {
+            $defLine->stmts = [];
+        }
+        $defText = $this->prettyPrinter->prettyPrint([$defLine]);
+        return strstr($defText, "\n", true) ?: $defText;
```

Comment:
`strstr` is **really** slow, why aren't you use `strpos`, which also consumes less-memory?